File: LeetCodeDSA/2611.py
def miceAndCheese(reward1, reward2, k):
    length = len(reward1)
    if k == length: 
        return sum(reward1)
    sum_mouse2 = sum(reward2)
    gain = [0] * length
    for i in range(length):
        gain[i] = reward1[i] - reward2[i]
    gain = sorted(gain, reverse=True)
    for i in range(k):
        sum_mouse2 += gain[i]
    return sum_mouse2, gain
    
    
        
print(miceAndCheese(reward1 = [1,1,3,4], reward2 = [4,4,1,1], k = 2))

File: LeetCodeDSA/2611.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>


int intComparator ( const void * first, const void * second ) {
    int firstInt = * (const int *) first;
    int secondInt = * (const int *) second;
    return secondInt - firstInt;
}


int main(){
    int reward1[] = {1,4,4,6,4}; 
    int reward1Size = 5;
    int reward2[] = {6,3,5,6,1}; 
    int reward2Size = 5;
    int k = 1;

    if(k == reward1Size){
        int total = 0; 
        for(int i = 0 ; i < reward1Size; i++){
            total += reward1[i];
        }
        printf("%d", total); 
        return 1;
    }
    int total = 0;
    for(int i =0;i < reward1Size; i++){
        total += reward2[i];
    }
    int gain[reward1Size];
    for(int i =0 ; i < reward1Size; i++){
        gain[i] = reward1[i] - reward2[i]; 
    }
    for(int i =0 ;i < reward1Size;i++){
        printf("%d ", gain[i]);
    }
    qsort(gain, reward1Size, sizeof(int),  intComparator);
    printf("\n");
    
    for(int i =0; i < k; i++){
        total += gain[i];
    }
    printf("%d", total);
    return 0;
}